nifi-toolkit/nifi-property-encryptor-tool/src/main/java/org/apache/nifi/util/file/ConfigurationFileUtils.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.util.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+public class ConfigurationFileUtils {
+
+    public static String DEFAULT_CONF_DIR = "conf";
+    public static String NIFI_PROPERTIES_DEFAULT_NAME = "nifi.properties";
+    public static String NIFI_REGISTRY_DEFAULT_PROPERTIES_NAME = "nifi-registry.properties";
+
+    public static File getTemporaryOutputFile(final String prefix, final File siblingFile) throws IOException {
+        if (siblingFile != null && siblingFile.isFile()) {
+            return File.createTempFile(prefix, siblingFile.getName(), siblingFile.getParentFile());
```

Comment:
The documentation for `java.io.File.createTempFile()` recommends using `java.nio.Files.createTempFile()` because it has more restrictive default permissions, so recommend changing the implementation to use `java.nio.Files.createTempFile()`.